prow/plugins/trigger/push.go

```diff
@@ -55,11 +57,33 @@ func createRefs(pe github.PushEvent) prowapi.Refs {
 	}
 }
 
+func getPullRequests(c githubClient, pe github.PushEvent) ([]github.PullRequest, error) {
+	org := pe.Repo.Owner.Login
+	repo := pe.Repo.Name
+	for _, commit := range pe.Commits {
+		if commit.ID == "" {
+			continue
+		}
+		pulls, err := c.ListCommitPullRequests(org, repo, commit.ID)
+		if err != nil {
+			return nil, err
+		}
+		if len(pulls) > 0 {
+			return pulls, nil
+		}
+	}
+	return nil, nil
+}
+
 func handlePE(c Client, pe github.PushEvent) error {
 	if pe.Deleted || pe.After == "0000000000000000000000000000000000000000" {
 		// we should not trigger jobs for a branch deletion
 		return nil
 	}
+	// TODO: it seems there is a bug in GH API, which returns an empty list of
+	// PRs for all commits listed by the push event. Waiting for 1 second seems
+	// to successfully work around this.
+	time.Sleep(1 * time.Second)
```

Comment:
Huh, this is really surprising. This looks like a possible source of surprising flakes (how do we know a second is always enough?) Can we get at least a confirmation that this is really a bug and this is a preferred way to work around it?

Reply:
Thanks for the review. I reported this to GitHub support and asked for confirmation and a work-around about the same time I submitted the PR (about 2 weeks ago). They have not confirmed yet, but based on their responses it does seem to be a bug.

Reply:
I have logged a bug with GitHub support team. Based on their feedback, they confirm this can happen but have not promised a fix. They agree the workaround I have used could work, but cannot guarantee that. The GH support ticket is titled "bug in endpoint for listing commit PRs", if further follow up on it is needed.